source/extensions/common/tap/tap_config_base.cc

```diff
@@ -277,6 +281,90 @@ void FilePerTapSink::FilePerTapSinkHandle::submitTrace(
   }
 }
 
+// The implemented code for UDP sink
+UdpTapSink::UdpTapSink(const envoy::extensions::tap_sinks::udp_sink::v3::UdpSink& config)
+    : config_(config) {
+  if (config_.udp_address().protocol() != envoy::config::core::v3::SocketAddress::UDP) {
+    ENVOY_LOG_MISC(warn, "{}: Only suport UDP and invalid protocol", __func__);
+    return;
```

Comment:
Will these early returns be leaving the object in an incorrect state?
What will happen if there's a warning (without initializing the data members), and then the extension is used for tracing?
Will it make sense to reject a bad configuration rather than accepting it and failing silently?

Reply:
>Will these early returns be leaving the object in an incorrect state?

Yes.

>What will happen if there's a warning (without initializing the data members), and then the extension is used for tracing?

Nothing, in UDP sink extension, it will check it  and return if object isn't full initialized

>Will it make sense to reject a bad configuration rather than accepting it and failing silently?

Yes, because for custom sink, it doesn't know which data is correct or not. this will be handle in specific sink, like UDP